ClosedXML/Excel/Ranges/XLRangeAddress.cs

```diff
@@ -367,6 +367,35 @@ public bool IsEntireSheet()
             return IsValid && IsEntireColumn() && IsEntireRow();
         }
 
+        public IXLRangeAddress Intersection(IXLRangeAddress otherRangeAddress)
+        {
+            if (otherRangeAddress == null)
+                throw new ArgumentNullException(nameof(otherRangeAddress));
+
+            var xlOtherRangeAddress = (XLRangeAddress)otherRangeAddress;
+            return Intersection(in xlOtherRangeAddress);
+        }
+
+        internal XLRangeAddress Intersection(in XLRangeAddress otherRangeAddress)
+        {
+            if (!this.Worksheet.Equals(otherRangeAddress.Worksheet))
+                throw new ArgumentOutOfRangeException(nameof(otherRangeAddress), "The other range address is on a different worksheet");
+
+            var thisRangeAddressNormalized = this.Normalize();
+            var otherRangeAddressNormalized = otherRangeAddress.Normalize();
+
+            var firstRow = Math.Max(thisRangeAddressNormalized.FirstAddress.RowNumber, otherRangeAddressNormalized.FirstAddress.RowNumber);
+            var firstColumn = Math.Max(thisRangeAddressNormalized.FirstAddress.ColumnNumber, otherRangeAddressNormalized.FirstAddress.ColumnNumber);
+            var lastRow = Math.Min(thisRangeAddressNormalized.LastAddress.RowNumber, otherRangeAddressNormalized.LastAddress.RowNumber);
+            var lastColumn = Math.Min(thisRangeAddressNormalized.LastAddress.ColumnNumber, otherRangeAddressNormalized.LastAddress.ColumnNumber);
+
+            return new XLRangeAddress
```

Comment:
You missed the validation 
```
if (lastRow < firstRow || lastColumn < firstColumn)
```
This should return an invalid address.

Consider the example:
```
var intersection = ws.Range("A1:C3").Intersection(ws.Range("A5:C8"));
Assert.IsFalse(intersection.IsValid);
```
Now the `intersection` is `A5:C3` which is wrong.

Reply:
Ok, I'll have to create an arbitrary invalid range address then. I'll just set all row and column numbers to -1. Agree?

Reply:
This is now done after the latest push. Added a static field `XLRangeAddress.Invalid` that we can reuse whenever we want to deliberately return an invalid range address.

Reply:
This is exactly what I was going to suggest :)